service/controller/v19/key/key.go

```diff
@@ -111,8 +112,12 @@ func AvailabilityZone(customObject v1alpha1.AWSConfig) string {
 	return customObject.Spec.AWS.AZ
 }
 
-func AvailabilityZones(customObject v1alpha1.AWSConfig) int {
-	return customObject.Spec.AWS.AvailabilityZones
+func AvailabilityZones(customObject v1alpha1.AWSConfig) []v1alpha1.AWSConfigStatusAWSAvailabilityZone {
+	zones := customObject.Status.AWS.AvailabilityZones
+	sort.Slice(zones, func(i, j int) bool {
+		return zones[i].Name < zones[j].Name
+	})
+	return zones
```

Comment:
When this is a thing the sorting should rather be done with the randomization directly in the ipam resource. When this is not a general thing we want it should may be only be done where we want exactly that behaviour.

Reply:
I moved sorting to site-of-use.